### PR TITLE
Bug 1960101: Fix update-codegen hack, pull in changes from openshift/api, bump k8.io deps to v0.22.1

### PR DESCRIPTION
**Pull request by astoycos**
Update ocp API to pull in @bpickard22's changes to netflow port validation 

Create new makefile step to make pulling in API changes much easier based on what is already done in https://github.com/openshift/cluster-ingress-operator

### CODE REVIEW

**Comment by astoycos** [comment]
test /ci/prow/e2e-openstack-ovn

**Comment by astoycos** [comment]
test ci/prow/e2e-openstack-ovn

**Comment by astoycos** [comment]
/assign @squeed @tssurya 

This is just a start, I just don't have time to add the verification step to CI... We should also eventually be generating/ validating `bindata` right?

**Comment by squeed** [comment]
@astoycos why not just merge this with `hack/update-generated.sh` and make a `hack/verify-generated.sh` while you're at it?

**Comment by astoycos** [comment]
Sounds good I can do that, thanks @squeed

**Comment by rbbratta** [comment]
Required for https://bugzilla.redhat.com/show_bug.cgi?id=196010  

We need to update the CRDs.

File: go.mod [hunk 1]
```diff
@@ -3,51 +3,53 @@ module github.com/openshift/cluster-network-operator
...
 	sigs.k8s.io/cluster-api-provider-openstack v0.3.3
 	sigs.k8s.io/controller-runtime v0.9.0-beta.0
-	sigs.k8s.io/structured-merge-diff/v4 v4.1.1 // indirect
 )
 
 replace (
-	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.2
-	google.golang.org/grpc => google.golang.org/grpc v1.27.1
-	k8s.io/apiserver => github.com/openshift/kubernetes-apiserver v0.0.0-20210419140141-620426e63a99 // points to temporary-watch-reduction-patch-1.21 to pick up k/k/pull/100959
+	github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.5.1
```
Comment (squeed) [inline go.mod:52]: Is this line still needed?

**Comment by squeed** [comment]
/approve
just one question.

File: go.mod (hunk 1 above)
Comment (astoycos) [reply go.mod:52]: Good question,  I don't think it is and can remove, but kube-openAPI (which is imported by `k8s.io/utils`) is what uses this library (https://github.com/kubernetes/kube-openapi/blob/master/go.mod#L15) and they have it bound there so I followed suite

Comment (astoycos) [reply go.mod:52]: However, I did just verify locally and everything seems to work, so I'll go ahead and remove.

Comment (squeed) [reply go.mod]: yah, the require makes sense, but the replace doesn't.

**Comment by squeed** [comment]
/lgtm
just needs a valid bz.

**Comment by astoycos** [comment]
Overriding `e2e-metal-ipi-ovn-ipv6 ` due to https://bugzilla.redhat.com/show_bug.cgi?id=2000086

**Comment by abhat** [comment]
/override ci/prow/e2e-gcp-ovn
Fails because of: https://bugzilla.redhat.com/show_bug.cgi?id=2000589

**Pull request merged**